python/lsst/pipe/drivers/constructCalibs.py

```diff
@@ -591,9 +618,18 @@ def scatterCombine(self, pool, outputId, ccdIdLists, scales):
         self.log.info("Scatter combination")
         outputIdItemList = outputId.items()
         data = [Struct(ccdIdList=ccdIdLists[ccdName], scales=scales[ccdName],
-                       outputId=dict(outputIdItemList +
-                                     [(k, ccdName[i]) for i, k in enumerate(self.config.ccdKeys)])) for
+                       outputId=dict([(k, ccdName[i]) for i, k in enumerate(self.config.ccdKeys)])) for
                 ccdName in ccdIdLists]
+        #
+        # Check if we need to look up any keys that aren't in the dataId
+        #
+        butler = pool._getCache(pool._context, 0).butler
```

Comment:
This is naughty (using not one implementation detail but two!). Can we do it another way, like passing the `butler` into this method explicitly? Or run the operation on the worker? Or get the right values in the `ccdName` in the first place?